File: src/main/java/io/myosotisdev/utopianism/modules/stat/data/EquipSlotData.java
package io.myosotisdev.utopianism.modules.stat.data;

import io.myosotisdev.utopianism.modules.item.ItemEquipSlot;
import io.myosotisdev.utopianism.serialize.EquipSlotTagSerializer;
import io.myosotisdev.utopianism.serialize.Serializer;

public class EquipSlotData implements ItemStatData<ItemEquipSlot>
{
    public static Serializer<ItemEquipSlot> Serializer = new EquipSlotTagSerializer();

    private ItemEquipSlot value;

    @Override
    public ItemEquipSlot getData()
    {
        return this.value;
    }
}

File: src/main/kotlin/io/myosotisdev/utopianism/modules/stat/data/EquipSlotData.kt
package io.myosotisdev.utopianism.modules.stat.data

import io.myosotisdev.utopianism.modules.item.ItemEquipSlot
import io.myosotisdev.utopianism.serialize.EquipSlotTagSerializer
import io.myosotisdev.utopianism.serialize.Serializer

class EquipSlotData(override var value: ItemEquipSlot = ItemEquipSlot.ALL) : ItemStatData<ItemEquipSlot>
{
    companion object
    {
        var Serializer: Serializer<ItemEquipSlot> = EquipSlotTagSerializer()
    }
}